File: src/main/java/indi/dkx/laocai/bot/model/segment/FaceSegment.java
package indi.dkx.laocai.bot.model.segment;

import lombok.Data;
import lombok.EqualsAndHashCode;
import indi.dkx.laocai.bot.model.segment.data.FaceSegmentData;

/**
 * face 表情消息段
 */
@EqualsAndHashCode(callSuper = true)
@Data
public class FaceSegment extends Segment<FaceSegmentData> {
    public FaceSegment(String faceId, boolean isLarge){
        super("face", new FaceSegmentData(faceId, isLarge));
    }
}

File: laocai-bot-core/src/main/kotlin/indi/kyson/laocai/bot/core/segment/FaceSegment.kt
package indi.kyson.laocai.bot.core.segment

import com.fasterxml.jackson.annotation.JsonIgnore

/**
 * 表情消息段。
 */
class FaceSegment private constructor(override val data: Data) : Segment {

    data class Data(val faceId: String, val isLarge: Boolean) {
        fun toSegment(): FaceSegment = FaceSegment(this)
    }

    override val type: String
        get() = "face"

    @get:JsonIgnore
    val faceId: String
        get() = data.faceId

    @get:JsonIgnore
    val isLarge: Boolean
        get() = data.isLarge

    companion object {
        @JvmStatic
        fun of(faceId: String, isLarge: Boolean): FaceSegment = FaceSegment(Data(faceId, isLarge))
    }
}
